pysam/libcalignmentfile.pyx

```diff
@@ -992,6 +997,16 @@ cdef class AlignmentFile(HTSFile):
                 if not self.is_stream:
                     self.start_offset = self.tell()
 
+        # Set any eventual extra threads
+        if n_threads > 1 and ignore_truncation:
+            # This won't raise errors if reaching a truncated alignment,
+            # because bgzf_mt_reader in htslib does not deal with
+            # bgzf_mt_read_block returning non-zero values, contrary
+            # to bgzf_read (https://github.com/samtools/htslib/blob/1.7/bgzf.c#L888)
+            # Better to avoid this (for now) than to produce seemingly correct results.
+            raise ValueError('Cannot add extra threads when "ignore_truncation" is True')
+        hts_set_threads(self.htsfile, n_threads - 1)
```

Comment:
Ideally, this call would be part of the `HTSFile` base object, so that `VariantFile` and other objects also benefit.  I very much like the conservative approach to interactions with truncation detection.  Also, please rename `n_threads` to `threads`, since we don't use the `n_` prefix elsewhere in the code.